src/request-manager.ts

```diff
@@ -0,0 +1,27 @@
+type RequestType = 'universalQuery'
+  | 'verticalQuery'
+  | 'universalAutoComplete'
+  | 'verticalAutoComplete';
+
+export default class RequestManager {
+  private latestRequestIds: Record<RequestType, number>;
+
+  constructor() {
+    this.latestRequestIds = {
+      universalQuery: 0,
+      verticalQuery: 0,
+      universalAutoComplete: 0,
+      verticalAutoComplete: 0
+    };
+  }
+
+  updateRequestId(requestName: RequestType): number {
+    const newId = ++this.latestRequestIds[requestName];
+    this.latestRequestIds[requestName] = newId;
+    return newId;
+  }
```

Comment:
We can simplify this function to just `return ++this.latestRequestIds[requestName];`